extensions/schema-compare/src/dialogs/schemaCompareDialog.ts

```diff
@@ -482,26 +479,51 @@ export class SchemaCompareDialog {
 
 		return {
 			component: this.sourceServerDropdown,
-			title: loc.ServerDropdownLabel
+			title: loc.ServerDropdownLabel,
+			actions: [this.sourceConnectionButton]
 		};
 	}
 
+	private createConnectionButton(isTarget: boolean): azdata.ButtonComponent {
+		const selectConnectionButton = this.view.modelBuilder.button().withProperties({
+			ariaLabel: loc.selectConnection,
+			iconPath: path.join(this.extensionContext.extensionPath, 'media', 'selectConnection.svg'),
+			height: '20px',
+			width: '20px'
+		}).component();
+
+		selectConnectionButton.onDidClick(async () => {
+			let connection = await azdata.connection.openConnectionDialog();
```

Comment:
What happens to source dropdown when I add connection through target dropdown or vice versa - is it added to both? Also, if so - take a second look that the oder of items in dropdown or selected item index behind the curtain does not get negatively affected. 

Is a test possible for this?

Reply:
The second gif that I shared adds connection though target dropdown and they do show up at source when database radio button is clicked.
But if the source dropdown is already populated, and a new connection is added in target, the new connection doesn't get auto populated in source. If either the radio button is toggled or connection button is clicked (though nothing new added)- that adds the new connection - it basically needs a trigger to capture all the active connections.

I will look adding a new test.

Reply:
can the source server dropdown's values be updated to include the new connection, while keeping its currently selected connection, when a new connection is added through to the target dropdown?

Reply:
Done.